src/material/table/_table-theme.scss

```diff
@@ -1,60 +1,44 @@
-@use 'sass:map';
-@use '@material/theme/theme-color' as mdc-theme-color;
-@use '@material/data-table/data-table' as mdc-data-table;
-@use '@material/data-table' as mdc-data-table-theme;
+@use '../core/tokens/m2/mat/table' as tokens-mat-table;
 @use '../core/theming/theming';
-@use '../core/mdc-helpers/mdc-helpers';
 @use '../core/typography/typography';
+@use '../core/tokens/token-utils';
 
-@mixin color($config-or-theme) {
-  $config: theming.get-color-config($config-or-theme);
-  // Save original values of MDC global variables. We need to save these so we can restore the
-  // variables to their original values and prevent unintended side effects from using this mixin.
-  $orig-selected-row-fill-color: mdc-data-table-theme.$selected-row-fill-color;
-  $orig-divider-color: mdc-data-table-theme.$divider-color;
-  $orig-row-hover-fill-color: mdc-data-table-theme.$row-hover-fill-color;
-  $orig-header-row-text-color: mdc-data-table-theme.$header-row-text-color;
-  $orig-row-text-color: mdc-data-table-theme.$row-text-color;
-  $orig-stroke-color: mdc-data-table-theme.$stroke-color;
-
-  @include mdc-helpers.using-mdc-theme($config) {
-    mdc-data-table-theme.$selected-row-fill-color: rgba(mdc-theme-color.prop-value(primary), 0.04);
-    mdc-data-table-theme.$divider-color: rgba(mdc-theme-color.prop-value(on-surface), 0.12);
-    mdc-data-table-theme.$row-hover-fill-color: rgba(mdc-theme-color.prop-value(on-surface), 0.04);
-    mdc-data-table-theme.$header-row-text-color: rgba(mdc-theme-color.prop-value(on-surface), 0.87);
-    mdc-data-table-theme.$row-text-color: rgba(mdc-theme-color.prop-value(on-surface), 0.87);
-    mdc-data-table-theme.$stroke-color: rgba(mdc-theme-color.prop-value(on-surface), 0.12);
-
-    @include mdc-data-table.table-styles($query: mdc-helpers.$mdc-theme-styles-query);
+@mixin _output-tokens {
```

Comment:
I think you can use this function instead: https://github.com/angular/components/blob/main/src/material/core/style/_sass-utils.scss#L11

Also, I'm not sure we need to clean it up, I'd like to move toward emitting the tokens under the user's selector (or document root if they don't specify one)